src/js/node/events.js

```diff
@@ -386,21 +388,20 @@ async function* on(emitter, event, options = {}) {
 
   for (const evName of options?.close || []) {
     emitter.on(evName, () => {
-      emitter.emit(event, undefined);
```

Comment:
Why is `undefined` emitted when the close event is called? I needed to remove this line after my changes or else I was stuck in an infinite loop after all line have been read-in.